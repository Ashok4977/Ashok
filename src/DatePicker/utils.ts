import deepmerge from 'deepmerge'
import {DatePickerConfiguration, isMultiSelection, isRangeSelection, Selection} from './types'
import {defaultConfiguration} from './DatePickerProvider'

export const sanitizeDate = (date: Date | string) => new Date(new Date(date).toDateString())

export const getInitialFocusDate = (selection?: Selection) => {
  if (!selection) return sanitizeDate(new Date())

  if (selection instanceof Date) {
    return sanitizeDate(selection)
  } else if (isMultiSelection(selection)) {
    return sanitizeDate(selection[0])
  } else if (isRangeSelection(selection)) {
    return sanitizeDate(selection.from)
  } else {
    return sanitizeDate(new Date())
  }
}

export const initializeConfigurations = (externalConfig?: DatePickerConfiguration) => {
  let sanitizedConfig = {}
  if (externalConfig) {
    sanitizedConfig = Object.entries(externalConfig).filter(item => item[1] !== undefined && item[1] !== null)
    for (const item of sanitizedConfig) {
      if (item instanceof Date) {
        sanitizeDate(item)
      }
    }
  }
  return deepmerge(defaultConfiguration, sanitizedConfig)
}
