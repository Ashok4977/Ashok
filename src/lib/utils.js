
/**
 * Splice items from an Array for with the test function returns
 * truthy, modifying the original list and returning an array of
 * the spliced items.
 *
 * ```js
 * const children = React.Children.toArray(props.children)
 * const buttons = splice(children, child => child.type === Button)
 * ```
 */
const splice = (list, test) => {
  return list.reduce((acc, item, i) => {
    if (test(item)) {
      return acc.concat(list.splice(i, 1))
    }
    return acc
  }, [])
}

export {splice}
