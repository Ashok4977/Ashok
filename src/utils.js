import chroma from 'chroma-js'
import {css} from 'styled-components'
import {theme} from 'styled-system'

// color shortcuts
export const color = name => theme(`colors.${name}`)
export const black = color('black')
export const white = color('white')

/**
 * Reduce a value to a primitive as long as it is a function.
 *
 * interpolate(() => 'x') // returns 'x'
 * interpolate(() => () => 'y') // returns 'y'
 */
export const interpolate = (value, context, props) => {
  while (typeof value === 'function') {
    value = value.call(context, props)
  }
  return value
}

/**
 * This decorator interpolates all of the function's arguments by calling any
 * functions in the provided context and with the provided arguments. In other
 * words, the function you provide is guaranteed to have all of its arguments
 * "called" if they are functions. Example:
 *
 * const rgba = interpolator((rgb, a) => chroma(rgb).alpha(a).css())
 * const fadeed = rgba(theme('colors.black'), .5) // returns a function!
 * faded({}) // probably throws an error because props.theme isn't set
 */
export const interpolator = f => {
  return (...outer) => {
    return function(props) {
      return f(...[...outer].map(a => interpolate(a, this, props)))
    }
  }
}

export const rgba = interpolator(       (rgb, alpha)    => chroma(rgb).alpha(alpha).css())
export const lighten = interpolator(    (value, amount) => chroma(value).brighten(amount).css())
export const darken = interpolator(     (value, amount) => chroma(value).darken(amount).css())
export const desaturate = interpolator( (value, amount) => chroma(value).desaturate(amount).css())
export const mix = interpolator(        (a, b, amount)  => chroma.mix(a, b, amount).css())
