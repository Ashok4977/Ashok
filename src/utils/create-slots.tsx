import React from 'react'
import useLayoutEffect from './useIsomorphicLayoutEffect'

/** createSlots is a factory that can create a
 *  typesafe Slots + Slot pair to use in a component definition
 *  For example: ActionList.Item uses createSlots to get a Slots wrapper
 *  + Slot component that is used by LeadingVisual, Description
 */
const createSlots = <SlotNames extends string>(_slotNames: SlotNames[]) => {
  type Slots = {
    [key in SlotNames]?: React.ReactNode
  }

  type ContextProps = {
    registerSlot: (name: SlotNames, contents: React.ReactNode) => void
    unregisterSlot: (name: SlotNames) => void
  }
  const SlotsContext = React.createContext<ContextProps>({
    registerSlot: () => null,
    unregisterSlot: () => null
  })

  const Slots: React.FC<{
    children: (slots: Slots) => React.ReactNode
  }> = ({children}) => {
    const [slots, setSlots] = React.useState<Slots>({})

    const registerSlot = React.useCallback(
      (name: SlotNames, contents: React.ReactNode) => setSlots(prevSlots => ({...prevSlots, [name]: contents})),
      []
    )

    // Slot can be removed from the tree as well,
    // we need to unregister them from the slot
    const unregisterSlot = React.useCallback(
      (name: SlotNames) =>
        setSlots(prevSlots => {
          const newSlots = {...prevSlots}
          delete newSlots[name]
          return newSlots
        }),
      []
    )

    return <SlotsContext.Provider value={{registerSlot, unregisterSlot}}>{children(slots)}</SlotsContext.Provider>
  }

  const Slot: React.FC<{
    name: SlotNames
    children: React.ReactNode
  }> = ({name, children}) => {
    const {registerSlot, unregisterSlot} = React.useContext(SlotsContext)

    useLayoutEffect(() => {
      registerSlot(name, children)
      return () => unregisterSlot(name)
    }, [name, children, registerSlot, unregisterSlot])

    return null
  }

  return {Slots, Slot}
}

export default createSlots
