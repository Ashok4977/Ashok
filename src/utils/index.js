
export function defined(val) {
  return val !== null && typeof val !== 'undefined'
}

export function definedFallback(input, fallback) {
  return defined(input) ? input : fallback
}
