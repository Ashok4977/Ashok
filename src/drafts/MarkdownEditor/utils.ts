import {isMacOS} from '@primer/behaviors/utils'

export type InputState = {
  readonly value: string
  readonly selectionStart: number
  readonly selectionEnd: number
}

export const getSelectedLineRange = (inputState: InputState): [number, number] => {
  // Subtract one from the caret position so the newline found is not the one _at_ the caret position
  // then add one because we don't want to include the found newline. Also changes -1 (not found) result to 0
  const start = inputState.value.lastIndexOf('\n', inputState.selectionStart - 1) + 1

  // activeLineEnd will be the index of the next newline inclusive, which works because slice is last-index exclusive
  let end = inputState.value.indexOf('\n', inputState.selectionEnd)
  if (end === -1) end = inputState.value.length

  return [start, end]
}

export const markdownComment = (text: string) => `<!-- ${text.replaceAll('--', '\\-\\-')} -->`

export const markdownLink = (text: string, url: string) =>
  `[${text.replaceAll('[', '\\[').replaceAll(']', '\\]')}](${url.replaceAll('(', '\\(').replaceAll(')', '\\)')})`

export const markdownImage = (altText: string, url: string) => `!${markdownLink(altText, url)}`

export const isModifierKey = (event: KeyboardEvent | React.KeyboardEvent<unknown>) =>
  isMacOS() ? event.metaKey : event.ctrlKey

export function isMultipleLines(string: string): boolean {
  return string.trim().split('\n').length > 1
}
