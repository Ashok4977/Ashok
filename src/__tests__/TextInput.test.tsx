import React from 'react'
import {TextInput} from '..'
import {render, mount, behavesAsComponent, checkExports} from '../utils/testing'
import {render as HTMLRender, cleanup, fireEvent} from '@testing-library/react'
import {axe, toHaveNoViolations} from 'jest-axe'
import 'babel-polyfill'
import {SearchIcon} from '@primer/octicons-react'
expect.extend(toHaveNoViolations)

describe('TextInput', () => {
  behavesAsComponent({Component: TextInput, options: {skipAs: true}})

  checkExports('TextInput', {
    default: TextInput
  })

  it('should have no axe violations', async () => {
    const {container} = HTMLRender(<TextInput aria-label="zipcode" name="zipcode" variant="small" />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
    cleanup()
  })

  it('renders', () => {
    expect(render(<TextInput name="zipcode" />)).toMatchSnapshot()
  })

  it('renders small', () => {
    expect(render(<TextInput name="zipcode" size="small" />)).toMatchSnapshot()
  })

  it('renders large', () => {
    expect(render(<TextInput name="zipcode" size="large" />)).toMatchSnapshot()
  })

  it('renders block', () => {
    expect(render(<TextInput name="zipcode" block />)).toMatchSnapshot()
  })

  it('renders warning', () => {
    expect(render(<TextInput name="zipcode" status="warning" />)).toMatchSnapshot()
  })

  it('renders error', () => {
    expect(render(<TextInput name="zipcode" status="error" />)).toMatchSnapshot()
  })

  it('renders contrast', () => {
    expect(render(<TextInput name="zipcode" status="contrast" />)).toMatchSnapshot()
  })

  it('renders placeholder', () => {
    expect(render(<TextInput name="zipcode" placeholder={'560076'} />)).toMatchSnapshot()
  })

  it('renders leadingVisual', () => {
    expect(render(<TextInput name="search" placeholder={'Search'} leadingVisual={SearchIcon} />)).toMatchSnapshot()
  })

  it('renders trailingVisual', () => {
    expect(render(<TextInput name="search" placeholder={'Search'} trailingVisual={SearchIcon} />)).toMatchSnapshot()
  })

  it('focuses the text input if you do not click the input element', () => {
    const {container, getByLabelText} = HTMLRender(
      <>
        {/* eslint-disable-next-line jsx-a11y/label-has-for */}
        <label htmlFor="testInput">Search</label>
        <TextInput id="testInput" name="search" placeholder={'Search'} trailingVisual={SearchIcon} />
      </>
    )

    const icon = container.querySelector('svg')!

    expect(getByLabelText('Search')).not.toEqual(document.activeElement)
    fireEvent.click(icon)
    expect(getByLabelText('Search')).toEqual(document.activeElement)
  })

  it('renders with a loading indicator', () => {
    expect(
      render(
        <>
          <TextInput isLoading />

          <TextInput isLoading loaderPosition="leading" />

          <TextInput isLoading loaderPosition="trailing" />

          <TextInput isLoading leadingVisual={SearchIcon} />

          <TextInput isLoading leadingVisual={SearchIcon} loaderPosition="leading" />

          <TextInput isLoading leadingVisual={SearchIcon} loaderPosition="trailing" />

          <TextInput isLoading trailingVisual={SearchIcon} />

          <TextInput isLoading trailingVisual={SearchIcon} loaderPosition="leading" />

          <TextInput isLoading trailingVisual={SearchIcon} loaderPosition="trailing" />

          <TextInput isLoading size="small" leadingVisual={SearchIcon} trailingVisual={SearchIcon} />

          <TextInput isLoading leadingVisual={SearchIcon} trailingVisual={SearchIcon} loaderPosition="leading" />

          <TextInput
            isLoading
            size="large"
            leadingVisual={SearchIcon}
            trailingVisual={SearchIcon}
            loaderPosition="trailing"
          />
        </>
      )
    ).toMatchSnapshot()
  })

  it('should call onChange prop with input value', () => {
    const onChangeMock = jest.fn()
    const component = mount(<TextInput onChange={onChangeMock} value="test" />)
    component.find('input').simulate('change')
    expect(onChangeMock).toHaveBeenCalled()
  })

  it('should render a password input', () => {
    expect(render(<TextInput name="password" type="password" />)).toMatchSnapshot()
  })
})
