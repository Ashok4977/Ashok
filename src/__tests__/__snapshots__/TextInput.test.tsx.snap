// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="zipcode"
    type="text"
  />
</span>
`;

exports[`TextInput renders block 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
  width: 100%;
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="zipcode"
    type="text"
  />
</span>
`;

exports[`TextInput renders consistently 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    type="text"
  />
</span>
`;

exports[`TextInput renders contrast 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="zipcode"
    type="text"
  />
</span>
`;

exports[`TextInput renders error 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
  border-color: #cf222e;
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

.c0:focus-within {
  border-color: #cf222e;
  box-shadow: 0 0 0 3px rgba(164,14,38,0.4);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="zipcode"
    type="text"
  />
</span>
`;

exports[`TextInput renders large 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="zipcode"
    type="text"
  />
</span>
`;

exports[`TextInput renders leadingIcon 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <svg
    aria-hidden="true"
    className="TextInput-leading-icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<path fill-rule=\\"evenodd\\" d=\\"M11.5 7a4.499 4.499 0 11-8.998 0A4.499 4.499 0 0111.5 7zm-.82 4.74a6 6 0 111.06-1.06l3.04 3.04a.75.75 0 11-1.06 1.06l-3.04-3.04z\\"></path>",
      }
    }
    fill="currentColor"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
  <input
    className="c1"
    data-component="input"
    name="search"
    placeholder="Search"
    type="text"
  />
</span>
`;

exports[`TextInput renders placeholder 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="zipcode"
    placeholder="560076"
    type="text"
  />
</span>
`;

exports[`TextInput renders small 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="zipcode"
    type="text"
  />
</span>
`;

exports[`TextInput renders trailingIcon 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="search"
    placeholder="Search"
    type="text"
  />
  <svg
    aria-hidden="true"
    className="TextInput-trailing-icon"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<path fill-rule=\\"evenodd\\" d=\\"M11.5 7a4.499 4.499 0 11-8.998 0A4.499 4.499 0 0111.5 7zm-.82 4.74a6 6 0 111.06-1.06l3.04 3.04a.75.75 0 11-1.06 1.06l-3.04-3.04z\\"></path>",
      }
    }
    fill="currentColor"
    height={16}
    role="img"
    style={
      Object {
        "display": "inline-block",
        "overflow": "visible",
        "userSelect": "none",
        "verticalAlign": "text-bottom",
      }
    }
    viewBox="0 0 16 16"
    width={16}
  />
</span>
`;

exports[`TextInput renders warning 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
  border-color: #bf8700;
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

.c0:focus-within {
  border-color: #bf8700;
  box-shadow: 0 0 0 3px rgba(212,167,44,0.4);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="zipcode"
    type="text"
  />
</span>
`;

exports[`TextInput should render a password input 1`] = `
.c1 {
  border: 0;
  font-size: inherit;
  font-family: inherit;
  background-color: transparent;
  -webkit-appearance: none;
  color: inherit;
  width: 100%;
}

.c1:focus {
  outline: 0;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-height: 34px;
  font-size: 14px;
  line-height: 20px;
  color: #24292f;
  vertical-align: middle;
  background-repeat: no-repeat;
  background-position: right 8px center;
  border: 1px solid #d0d7de;
  border-radius: 6px;
  outline: none;
  box-shadow: inset 0 1px 0 rgba(208,215,222,0.2);
  cursor: text;
  padding: 6px 12px;
  display: grid;
  grid-template-areas: 'leadingIcon input trailingIcon';
}

.c0 >:not(:last-child) {
  margin-right: 8px;
}

.c0 [data-component='input'] {
  grid-area: input;
}

.c0 .TextInput-leading-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: leadingIcon;
}

.c0 .TextInput-trailing-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #57606a;
  grid-area: trailingIcon;
}

.c0:focus-within {
  border-color: #0969da;
  box-shadow: 0 0 0 3px rgba(9,105,218,0.3);
}

@media (min-width:768px) {
  .c0 {
    font-size: 14px;
  }
}

<span
  className="c0 TextInput-wrapper"
>
  <input
    className="c1"
    data-component="input"
    name="password"
    type="password"
  />
</span>
`;
